w09h03/test/pgdp/filter/FrameIteratorTest.java

```diff
@@ -0,0 +1,126 @@
+package pgdp.filter;
+
+import org.bytedeco.javacv.FFmpegFrameGrabber;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Assertions;
+
+import java.awt.image.BufferedImage;
+import java.io.FileNotFoundException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class FrameIteratorTest {
+
+  private static Iterator<Frame> getFrameIteratorFromContainer(VideoContainer con) {
```

Comment:
This is buggy. At least in my implementation this gets called after the first element of the iterator already got consumed and therefore leads to weird results with `testFrameIterator_multipleHasNextCalls()` and `testFrameIterator_onlyNext()`.

Reply:
Huh? So your constructor of VideoContainer already fetches the first frame?

Reply:
Exactly